pkg/util/kernel/find_headers.go

```diff
@@ -148,25 +141,50 @@ func validateHeaderDirs(hv Version, dirs []string) []string {
 			log.Debugf("error validating %s: error validating headers version: %w", d, err)
 			continue
 		}
-		if dirv != hv {
+
+		if !compareKernelVersionsRelaxed(dirv, hv) {
```

Comment:
Do you have something I can read regarding this? We are using the actual `LINUX_VERSION_CODE` constant from the `version.h` file, which should contain the sublevel.

Reply:
Yes this is a bit strange but on OpenSUSE 15:
`uname -a` = `Linux opensuse15 5.3.18-59.37-default #1 SMP Mon Nov 22 12:29:04 UTC 2021 (d10168e) x86_64 x86_64 x86_64 GNU/Linux`
```
>> cat /usr/src/linux-5.3.18-59.37-obj/x86_64/default/include/generated/uapi/linux/version.h
#define LINUX_VERSION_CODE 328466
#define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
```

```
>> cat /usr/include/linux/version.h
#ifdef __KERNEL__
#error "======================================================="
#error "You should not include /usr/include/{linux,asm}/ header"
#error "files directly for the compilation of kernel modules."
#error ""
#error "glibc now uses kernel header files from a well-defined"
#error "working kernel version (as recommended by Linus Torvalds)"
#error "These files are glibc internal and may not match the"
#error "currently running kernel. They should only be"
#error "included via other system header files - user space"
#error "programs should not directly include <linux/*.h> or"
#error "<asm/*.h> as well."
#error ""
#error "Since Linux 2.6, the kernel module build process has been"
#error "updated such that users building modules should not typically"
#error "need to specify additional include directories at all."
#error ""
#error "To build kernel modules, ensure you have the build environment "
#error "available either via the kernel-devel and kernel-<flavor>-devel "
#error "packages or a properly configured kernel source tree."
#error ""
#error "Then, modules can be built using:"
#error "make -C <path> M=$PWD"
#error ""
#error "For the currently running kernel there will be a symbolic "
#error "link pointing to the build environment located at "
#error "/lib/modules/$(uname -r)/build for use as <path>."
#error ""
#error "If you are seeing this message, your environment is "
#error "not configured properly. "
#error ""
#error "Please adjust the Makefile accordingly."
#error "======================================================="
#else
#define LINUX_VERSION_CODE 328448
#define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
#endif
```

Where `328466 = 5.3.18` and `328448 = 5.3.0`

Reply:
Let's keep `/usr` and a full version check for now. If in the future we see some issues on OpenSUSE we'll find a better solution to this issue

Reply:
I say this because `kconfig.h` is not in `/usr` so it should be removed by the critical headers check